src/libraries/System.Memory.Data/ref/System.Memory.Data.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the https://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+namespace System
+{
+    public partial class BinaryData
+    {
+        public BinaryData(byte[] data) { }
+        public BinaryData(object? jsonSerializable, System.Text.Json.JsonSerializerOptions? options = null, System.Type? type = null) { }
+        public BinaryData(System.ReadOnlyMemory<byte> data) { }
+        public BinaryData(string data) { }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object? obj) { throw null; }
+        public static System.BinaryData FromBytes(byte[] data) { throw null; }
+        public static System.BinaryData FromBytes(System.ReadOnlyMemory<byte> data) { throw null; }
+        public static System.BinaryData FromObjectAsJson<T>(T jsonSerializable, System.Text.Json.JsonSerializerOptions? options = null) { throw null; }
+        public static System.BinaryData FromStream(System.IO.Stream stream) { throw null; }
+        public static System.Threading.Tasks.Task<System.BinaryData> FromStreamAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
How come this isn't ValueTask?

Reply:
This was discussed over at https://github.com/Azure/azure-sdk-for-net/pull/16248, but I can't find the comment right now because of a rebase. If I recall, the pattern was that the `ValueTask<T>` APIs were moved to extension methods, allowing this method to be a simple wrapper over the existing Stream methods.